src/uk/gov/hmcts/contino/DocumentPublisher.groovy

```diff
@@ -35,19 +34,29 @@ class DocumentPublisher implements Serializable {
     publish(getCosmosDbKey(), collectionLink, documents)
   }
 
-  @NonCPS
   private def publish(cosmosDbKey, collectionLink, documents) {
-
     def cosmosDbUrl = params.subscription == 'sandbox' ? DB_SANDBOX_URL : DB_DEFAULT_URL
-    def documentClient = new DocumentClient(cosmosDbUrl, cosmosDbKey, null, null)
+    def collection = collectionLink.split('/').last()
 
-    try {
-      documents.each {
-        documentClient.createDocument(collectionLink, new Document(it), null, false)
-      }
-    }
-    finally {
-      documentClient.close()
+    steps.sh "mkdir -p /tmp/metrics-reporting"
```

Comment:
Any issue with concurrency here? If multiple builds are using the same directory at the same time?

Reply:
we only run one executor so it's fine